### PR TITLE
✨ vGPU implementation - follow-ups

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:
Follow-up to #3025

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by sbueringer** [comment]
Will rebase once #3025 merges

**Review by neolit123 — COMMENTED** [review]
/lgtm

File: apis/v1beta1/types.go [hunk 1]
```diff
@@ -253,21 +253,21 @@ type PCIDeviceSpec struct {
...
-	// +kubebuilder:validation:Required
+	// +optional
 	VendorID *int32 `json:"vendorId,omitempty"`
 	// VGPUProfile is the profile name of a virtual machine's vGPU, in string.
 	// Defaults to the eponymous property value in the template from which the
 	// virtual machine is cloned.
 	// Mutually exclusive with DeviceID and VendorID as VGPUProfile and DeviceID + VendorID
 	// are two independent ways to define PCI devices.
-	// +kubebuilder:validation:Required
+	// +optional
```
Comment (neolit123) [inline apis/v1beta1/types.go:270]: makes sense

Comment (sbueringer) [reply apis/v1beta1/types.go:270]: I was confused at first why the field was not required in the CRD, but then I realized that omitempty overwrites the explicit marker..

Comment (neolit123) [reply apis/v1beta1/types.go:270]: what made sense to me is that deviceID + vendorID are mut.ex. with VGPUProfile, if i remember the original PR logic.

Comment (sbueringer) [reply apis/v1beta1/types.go:270]: Yup, having them all as optional and then having validation in the webhook definitely makes sense

**Pull request merged**